src/domain/Tables/LegacyDataTable/DataTablePagination.tsx

```diff
@@ -255,28 +208,75 @@ export class DataTablePagination extends React.Component<IDataTablePaginationPro
   public render () {
     const {
       className,
-      customComponent
+      customComponent,
+      showPageSizeOptions,
+      pageSizeOptions,
+      pageSize,
+      onPageSizeChange,
+      totalCount
     } = this.props
 
     return (
       <div className={classNames(className, '-pagination')} >
         {customComponent}
-        {this.pageSizeOptions}
-        {this.pageDetails}
+        <PageSizeOptions
+          showPageSizeOptions={showPageSizeOptions}
+          pageSizeOptions={pageSizeOptions}
+          pageSize={pageSize}
+          onPageSizeChange={onPageSizeChange}
+          page={this.state.page}
+        />
+        <PageDetails
+          pageSize={pageSize}
+          totalCount={totalCount}
+          page={this.state.page}
+        />
         {this.paginationComponents}
       </div>
     )
   }
+}
 
-  private changeHandler = (e: React.ChangeEvent<HTMLSelectElement>) => {
-    const {
-      onPageSizeChange
-    } = this.props
+const PageDetails: React.FC<any> = ({pageSize, totalCount, page}) => {
+  const t = useTranslateFunction()
 
-    const {
-      page
-    } = this.state
+  const entryCount = totalCount
+  const nextPage = page + 1
+  const maxPossibleCurrentPageIndex = nextPage * pageSize
+  const maxActualCurrentPageIndex = Math.min(entryCount, maxPossibleCurrentPageIndex)
+  const pageStartingIndex = page * pageSize || 1
+
+  return (
+    <div className='page-details'>
+      {t('pagination.details', {firstItemOnPage: pageStartingIndex, lastItemOnPage: maxActualCurrentPageIndex, totalItems: entryCount})}
+    </div>
+  )
+}
+
+const PageSizeOptions: React.FC<any> = ({showPageSizeOptions, pageSizeOptions, pageSize, onPageSizeChange, page}) => {
+  const t = useTranslateFunction()
+
+  if (showPageSizeOptions) {
+    const pageSizeSelect = (
+      <select
+        // tslint:disable-next-line:jsx-no-lambda
+        onChange={(e) => onPageSizeChange(Number(e.target.value), page)}
+        value={pageSize}
+      >
+        {pageSizeOptions.map((option: number) => (
+          <option key={option} value={option}>
+            {option}
+          </option>
+        ))}
+      </select>
+    )
 
-    return onPageSizeChange(Number(e.target.value), page)
+    return (
+      <span className='page-size-options'>
+        <label>{pageSizeSelect} {t('pagination.entriesPerPage')}</label>
```

Comment:
Is this right? Before `pageSizeSelect` was wrapped by text

Reply:
I had to change this because it wouldn't work as an i18n string. `pageSizeSelect` is literally a `<select>` element which we can't (or if we actually can, we shouldn't)  pass as a variable to the i18n.